dotfile/src/lib.rs

```diff
@@ -29,13 +36,82 @@ impl SetupAdapter for DotfileSetup {
         &self,
         config_string: &str,
         config_extension: &str,
-        _base_config: &BasicConfigContent,
+        base_config: &BasicConfigContent,
     ) -> Result<()> {
         let config: DotfileConfiguration =
             core::configuration::generate_config(config_string, config_extension)?;
 
-        println!("{config:?}");
+        config.create_files(base_config);
+        config.create_program_files(base_config)?;
+
+        Ok(())
+    }
+}
+
+impl DotfileConfiguration {
+    fn create_files(&self, base_config: &BasicConfigContent) {
+        self.files
+            .iter()
+            .flatten()
+            .for_each(|f| f.create_system_file(base_config));
+    }
+
+    fn create_program_files(&self, base_config: &BasicConfigContent) -> Result<()> {
+        let nvim = match &self.nvim {
+            Some(nvim) => nvim,
+            None => return Ok(()),
+        };
+
+        let target_path = format!(
+            "{}/{}",
+            &base_config
+                .base
+                .base_path
+                .to_str()
+                .expect("this should have already been validated to exist."),
+            GitModule::get_project_path(nvim, base_config)
+                .expect("Malformed git resource mentioned!"),
+        );
+
+        GitModule::git_clone(nvim, &target_path)?;
 
         Ok(())
     }
 }
+
+impl DotfileDefinition {
+    fn create_system_file(&self, base_config: &BasicConfigContent) {
+        if let Some(parent) = self.path.parent() {
+            let _ = fs::create_dir_all(parent);
+        }
+
+        let mut f = match base_config.force {
+            true => match File::create(&self.path) {
+                Ok(f) => f,
+                Err(err) => {
+                    println!("{err:?}");
+
+                    return;
+                }
+            },
+            false => match File::create_new(&self.path) {
+                Ok(f) => f,
+                Err(err) => {
+                    match err.kind() {
+                        ErrorKind::AlreadyExists => println!(
+                            "{:?} already exists. Delete or backup the current version and try again.",
+                            self.path
+                        ),
+                        _ => println!("{err:?}"),
+                    }
+
+                    return;
+                }
+            },
+        };
```

Comment:
A couple things to say here:

1. Generally a bad idea to use `match` on a `boolean`. Just use `if/else`
2. Generally a bad idea to use `match` on a `Result` unless you're selecting for different errors. You should usually map.

```suggestion
        let mut f = if base_config.force {
            File::create(&self.path)?
        } else { 
            File::create_new(&self.path)
                .map_err(|err| {
                    match err.kind() {
                        ErrorKind::AlreadyExists => println!(
                            "{:?} already exists. Delete or backup the current version and try again.",
                            self.path
                        ),
                        _ => println!("{err:?}"),
                    };
                    err
                })?
        };
```

again, i don't love using print inside this -- I think you'd be better off logging or mapping the error to an anyhow error